userprofile/templates/userprofile/profile.html

```diff
@@ -2,96 +2,120 @@
+			<input readonly type="text" class="form-control read-only" id="api-access-token"
+				value="{{ profile.api_token }}">
+			<span class="input-group-btn">
+				<button data-toggle="tooltip" data-placement="bottom"
+					title="{% trans 'Copy API Access Token' %}" aria-label="{% trans 'Copy API Access Token' %}"
+					class="aplus-button--secondary aplus-button--md js-copy" type="button"><span
+						class="glyphicon glyphicon-copy"></span><span class="">{% trans 'Copy' %}</span></button>
+			</span>
+		</div>
+		<br>
+		<div>
+			<label for="local-storage-services">{% trans "Automatically log in to:" %}</label>
+			<p id="services-instructions">
+				{% blocktrans trimmed%}
+				These are all services that you have marked for automatic redirection on this device. You can remove
+				automatic redirection here and you will be shown the privacy information next time you are being
+				redirected to the service.
 				{% endblocktrans %}
 			</p>
-			<ul class="local-storage-fields" data-forget-text="{% trans "Forget automatic redirection" %}"></ul>
-			</dd>
+			<ul id="local-storage-services" class="local-storage-fields"
+			data-forget-text="{% trans 'Forget automatic redirection' %}" role="group"></ul>
 		</div>
-    </div>
-    <div class="col-md-4">
-
-        {# TODO: list enrolled courses #}
-    </div>
+		<form action="{% url 'set-user-language' %}" method="post" class="form form-horizontal">{% csrf_token %}
+			<div class="form-group">
+				<label class="col-xs-12" for="language-profile">{% trans 'Language' %}</label>
+				<div class="col-xs-12 col-sm-8">
+					<select name="language" class="form-control" id="language-profile">
+						{% get_current_language as LANGUAGE_CODE %}
+						{% get_available_languages as LANGUAGES %}
+						{% get_language_info_list for LANGUAGES as languages %}
+						{% for language in languages %}
+						<option value="{{ language.code }}" {% if language.code == LANGUAGE_CODE %}selected{% endif %}>
+							{{ language.name_local }}
+						</option>
+						{% endfor %}
+					</select>
+				</div>
+				<div class="col-xs-offset-0 col-xs-6 col-sm-4 ">
+					<button type="submit"
+						class="aplus-button--default aplus-button--md btn-block btn-save-profile">{% trans 'Save' %}</button>
+				</div>
+			</div>
+		</form>
+	</div>
 </div>
+<div class="row">
+	<div class="col-sm-offset-2 col-sm-8 col-md-4">
+		{% if not is_google %}
+			{% brand_university_name as university %}
+				<h2>{% blocktrans with university=university trimmed %}Information provided by {{ university }}{% endblocktrans %}</h2>
+		{% elif is_google %}
+			<h2>{% trans "Information provided by Google:" %}</h2>
+		{% endif %}
+		<p class="text-warning">{% trans "The following settings should be changed in your identity provider" %}.</p>
+		<dl class="dl-horizontal">
+			{% trans "[Missing]" as miss %}
+			<dt class="provider-information">{% trans "Student ID" %}</dt>
+			<dd>{% firstof profile.student_id miss %}</dd>
```

Comment:
Note that the student ID is always empty when Google login is used and it can not be set in Google. Maybe that is clear enough even though the page says that the info "should be changed at Google".

Reply:
Or maybe we can add the student id to the header.
![image](https://user-images.githubusercontent.com/10437475/96690602-8f922180-138c-11eb-99f1-25fcc33e3f08.png)

Reply:
Interesting idea. I would now keep the student id under the institution section since the student id really comes from the institution for Aalto students. Do you strongly feel like moving the student id to the top?

Reply:
For me, it looks nice, and it could solve the confusion when the students log in with a Google account, but I do not have a strong opinion about this. We can also leave it where it is now.